packages/TypeDeclaration/src/Rector/Property/PropertyTypeDeclarationRector.php

```diff
@@ -57,7 +59,12 @@ public function refactor(Node $node): ?Node
         }
 
         $type = $this->propertyTypeInferer->inferProperty($node);
-        if ($type instanceof MixedType) {
+        /** @var Doc $attribute */
+        $attribute = $node->getAttribute('comments')[0];
+        $attributeText = $attribute !== null ? $attribute->getText() : '';
+        $isInheritdoc = Strings::contains(Strings::lower($attributeText), '@inheritdoc');
+
+        if ($type instanceof MixedType || $isInheritdoc) {
             return null;
         }
 
```

Comment:
This logic should be extracted to method in `DocBlocManipulator`, as that's a service that handles all around docblocks.

Reply:
👍